### PR TITLE
Smoke_Rebuild_Butts_Smokeable

### PR DESCRIPTION
**Pull request by karols146**
Taking ideas from #15949, Not finished yet, need to set up morale boost,and addition reactions. 
Just showing now and waiting for comments, what I should fix etc.

### CODE REVIEW

File: data/json/items/comestibles.json [hunk 1]
```diff
@@ -10454,32 +10454,36 @@
...
+        "type" : "GENERIC",
+        "id" : "cigar_butt_lit",
+        "name" : "cigar butt",
+        "weight" : "0",
+        "volume" : "0",
+        "color" : "dark_gray",
+        "symbol" : "!",
+        "description" : "A true gentleman always finishes his cigars. Always...",
+		"category" : "drugs",
+        "flags" : ["TABACCO","LITCIG","LIGHT_2"]
```
Comment (Coolthulhu) [inline data/json/items/comestibles.json]: Typo in tobacco

File: data/json/items/comestibles.json [hunk 2]
```diff
@@ -10454,32 +10454,36 @@
...
+	{
+        "type" : "GENERIC",
+        "id" : "cigar_butt_lit",
+        "name" : "cigar butt",
+        "weight" : "0",
+        "volume" : "0",
+        "color" : "dark_gray",
+        "symbol" : "!",
+        "description" : "A true gentleman always finishes his cigars. Always...",
+		"category" : "drugs",
```
Comment (Coolthulhu) [inline data/json/items/comestibles.json]: Don't use tabs, only spaces

File: src/iuse.cpp [hunk 3]
```diff
@@ -544,6 +554,14 @@ int iuse::smoking(player *p, item *it, bool, const tripoint& )
...
             p->set_painkiller( ( p->get_painkiller() + 3 ) * 2 );
         }
+	}	else if (it->type->id == "joint_roach") {
+        cig = item("joint_roach_lit", int(calendar::turn));
+        cig.item_counter = 10;
+        p->mod_hunger(4);
+        p->mod_thirst(6);
+        if( p->get_painkiller() < 5 ) {
+            p->set_painkiller( ( p->get_painkiller() + 3 ) * 2 );
+        }
```
Comment (Coolthulhu) [inline src/iuse.cpp]: Same effects as the whole item?

Comment (karols146) [inline src/iuse.cpp]: Nerf about half would be good ?

File: data/json/items/comestibles.json (hunk 2 above)
Comment (karols146) [inline data/json/items/comestibles.json]: Should look good now, have no idea why it didn't change yet.

File: data/json/items/generic.json [hunk 4]
```diff
@@ -2339,47 +2339,68 @@
...
         "flags" : [ "LITCIG", "LIGHT_4" ]
     },
     {
-        "type" : "GENERIC",
+        "type" : "COMESTIBLE",
         "id" : "joint_roach",
+		"use_action" : "SMOKING",
         "name" : "joint roach",
         "name_plural": "joint roaches",
+		"comestible_type" : "MED",
```
Comment (Coolthulhu) [inline data/json/items/generic.json]: Tabs here and above

**Comment by Coolthulhu** [comment]
What editor are you using? Notepad++ is good if you're on Windows.

Most text editors support searching for tabs. You usually have to type them out as \t (backslash and t). Notepad++ also supports replacing all tabs in file or selection with spaces.

**Comment by karols146** [comment]
I just try to learn vim for now.

File: data/json/items/generic.json [hunk 5]
```diff
@@ -2339,47 +2339,68 @@
...
     },
     {
         "type" : "GENERIC",
+        "id" : "cig_butt_lit",
+        "name" : "cigarette butt",
+        "weight" : "0",
+        "volume" : "0",
+        "color" : "dark_gray",
+        "symbol" : "!",
+        "description" : "Smelly piece of trash. You must be very desperate if you even try to smoke it.",
```
Comment (Night-Pryanik) [inline data/json/items/generic.json]: Double space please.

File: data/json/items/generic.json [hunk 6]
```diff
@@ -2339,47 +2339,68 @@
...
+
+    },
+    {
+        "type" : "GENERIC",
+        "id" : "joint_roach_lit",
+        "name" : "joint roach",
+        "name_plural": "joint roaches",
+        "color" : "green",
+        "symbol" : "!",
+        "description" : "The smoked-down butt of a joint, a reminder of some good times.  Why you smoke it again ?.",
```
Comment (Night-Pryanik) [inline data/json/items/generic.json]: Excess space before question mark. Also, excess period after question mark.
Also, "Why you smoke it again?" is unfinished sentence. Maybe "Why do you want to smoke it again?"

File: data/json/items/generic.json [hunk 7]
```diff
@@ -2339,47 +2339,68 @@
...
         "type" : "GENERIC",
+        "id" : "cig_butt_lit",
+        "name" : "cigarette butt",
+        "weight" : "0",
+        "volume" : "0",
+        "color" : "dark_gray",
+        "symbol" : "!",
+        "description" : "Smelly piece of trash. You must be very desperate if you even try to smoke it.",
+        "category" : "drugs",
+        "flags" : ["TABACCO","LITCIG","LIGHT_2"]
```
Comment (Night-Pryanik) [inline data/json/items/generic.json]: TABACCO -> TOBACCO.

File: data/json/items/generic.json [hunk 8]
```diff
@@ -1527,52 +1527,52 @@
...
-    "description": "This memory card appears to be related to 'XEDRA', and is certainly encrypted.  Looks * Interesting *, though...",
-    "price": 10000,
-    "price_postapoc": 0,
-    "material": "plastic",
-    "flags" : ["MC_MOBILE", "MC_ENCRYPTED", "MC_TURN_USED", "MC_SCIENCE_STUFF"],
-    "weight": 5,
-    "volume": 0
-  },
+    },
+    {
```
Comment (illi-kun) [inline data/json/items/generic.json]: This formatting is not related to the main aim of this PR. It makes your PR bigger and excessively hard to review with no actual reason.

**Comment by illi-kun** [comment]
I don't think that general idea of that PR is have to be implemented. What it adds to the game? Previously, it was clear path from something useful (cigarette) to something almost useless (cigarette butt, essentially a trash but can be utilized in something useful if you ready to collect some trash for a minor benefit).

As far as I understood your PR, now an extra step is added to this consumption chain, so now it's cig -> cig butt (-> cig butt butt??). Why it's useful? Why the consumer doesn't smoke it to the unsmokable condition at first attempt? What it brings to the game except excessive keypresses and hidden mechanic (because not all players will even try to smoke it)?

**Comment by karols146** [comment]
illi-kun, You are right, players don't even know that you can smoke that again, and Why you just want to smoke it, after night i just see this in other way. Ok so no problem I just close this and start looking about small fire little nerf. 
Thanks for your time to revive this guys, we got somewhere example .json file, just to be sure how it should look, and don't make this silly mistakes.
